From d5d56bb21fcde74999de2f99b68504b2871eb328 Mon Sep 17 00:00:00 2001
From: Tejun Heo <tj@kernel.org>
Date: Fri, 23 Feb 2024 09:40:31 -1000
Subject: [PATCH 139/149] scx: Initialize p->scx.runnable_at to jiffies instead
 of INITIAL_JIFFIES

INITIAL_JIFFIES is used to force overflow soon after boot to verify wrap
handling and makes no sense for jiffy values which can be set while the
system is running. Let's initialize p->scx.runnable_at to jiffies instead of
INITIAL_JIFFIES. Note this doesn't affect actual operation as the value will
be set when the task becomes ready.
---
 kernel/sched/core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 66f8b5377..6da5eb1e0 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4586,7 +4586,7 @@ static void __sched_fork(unsigned long clone_flags, struct task_struct *p)
 	p->scx.kf_mask		= 0;
 	atomic_long_set(&p->scx.ops_state, 0);
 	INIT_LIST_HEAD(&p->scx.runnable_node);
-	p->scx.runnable_at	= INITIAL_JIFFIES;
+	p->scx.runnable_at	= jiffies;
 	p->scx.ddsp_dsq_id	= SCX_DSQ_INVALID;
 	p->scx.ddsp_enq_flags	= 0;
 	p->scx.slice		= SCX_SLICE_DFL;
-- 
2.43.0.232.ge79552d197

